File: CustomisedCollections/src/dsa/MyStack.java
package dsa;

import java.util.EmptyStackException;

public class MyStack {
    private int count;
    private String[] elements;

    public MyStack(int maxSize) {
        elements = new String[maxSize];
    }

    public boolean isEmpty(){
        if(count == 0) return true;
        return false;
    }

    public void push(String element) {
        if(!isFull()) elements[count++] = element;
        else throw new IllegalStateException("Full!");
    }

    public String pop(){
        if(!isEmpty()) return elements[--count];
        else throw new EmptyStackException();
    }

    public String peek() {
        if(isEmpty()) throw new EmptyStackException();
        return elements[count - 1];
    }

    public boolean isFull() {
        if(count == elements.length) return true;
        return false;
    }
}

File: CustomisedCollections/src/dsa/my_stack.py
class MyStack:
    def __init__(self, maxSize):
        self.stack = []
        self.count = 0
        self.maxsize = maxSize

    def push(self,item):
        if not(self.is_full()):
            self.stack.append(item)
            self.count += 1
        else:
            raise OverflowError("Stack is full")

    def pop(self):
        if self.count > 0:
            element = self.stack[self.count-1]
            self.count -= 1
            return element
        else:
            raise OverflowError("Stack is empty")

    def is_empty(self):
        if self.count == 0:
            return True
        else:
            return False

    def peek(self):
        if self.count > 0:
            return self.stack[self.count-1]
        else:
            raise OverflowError("Stack is empty")

    def is_full(self):
        if self.count == self.maxsize:
            return True
        else:
            return False